src/test/java/io/managed/services/test/rhoas/KafkaRhoasAclTests.java

```diff
@@ -0,0 +1,245 @@
+        cli.listKafka();
+
+        // create kafka instance
+        LOGGER.info("create kafka instance with name {}", KAFKA_INSTANCE_NAME);
+        var k = cli.createKafka(KAFKA_INSTANCE_NAME);
+        LOGGER.debug(k);
+        LOGGER.info("wait for kafka instance: {}", k.getId());
+        KafkaRequest kafka = CLIUtils.waitUntilKafkaIsReady(cli, k.getId());
+        LOGGER.debug(kafka);
+    }
+
+    @AfterClass(alwaysRun = true)
+    @SneakyThrows
+    public void clean() {
+        var kafkaMgmtApi =  KafkaMgmtApiUtils.kafkaMgmtApi(Environment.OPENSHIFT_API_URI, Environment.PRIMARY_OFFLINE_TOKEN);
+        var securityMgmtApi = SecurityMgmtAPIUtils.securityMgmtApi(Environment.OPENSHIFT_API_URI, Environment.PRIMARY_OFFLINE_TOKEN);
+        
+        if (Environment.SKIP_KAFKA_TEARDOWN) {
+            LOGGER.warn("skip kafka instance clean up");
+            return;
+        }
+
+        try {
+            KafkaMgmtApiUtils.deleteKafkaByNameIfExists(kafkaMgmtApi, KAFKA_INSTANCE_NAME);
+        } catch (Throwable t) {
+            LOGGER.error("delete kafka instance error: ", t);
+        }
+
+        try {
+            SecurityMgmtAPIUtils.cleanServiceAccount(securityMgmtApi, SERVICE_ACCOUNT_NAME);
+        } catch (Throwable t) {
+            LOGGER.error("delete service account error: ", t);
+        }
+
+        try {
+            LOGGER.info("logout user from rhoas");
+            cli.logout();
+        } catch (Throwable t) {
+            LOGGER.error("CLI logout error: ", t);
+        }
+
+        try {
+            LOGGER.info("delete workdir: {}", cli.getWorkdir());
+            FileUtils.deleteDirectory(new File(cli.getWorkdir()));
+        } catch (Throwable t) {
+            LOGGER.error("clean workdir error: ", t);
+        }
+
+        bwait(vertx.close());
+    }
+    
+    @DataProvider(name = "aclEntityPairs")
+    public Object[][] aclEntityPairsDataProvider() {
+        return new Object[][]{
+            {ACLEntityType.USER, EXTERNAL_USER},
+            {ACLEntityType.SERVICE_ACCOUNT, SERVICE_ACCOUNT_NAME}
+        };
+    }
+
+    // Primary user creates specific ACL (details not important) which allows entity to perform all operation upon specific topic
+    // Outcome: ACL assigned to entity is created and can be found once it is searched for based on entity
+    @SneakyThrows
+    @Test(dataProvider = "aclEntityPairs", priority = 1)
+    public void testCreateAllAllowAcl(ACLEntityType entityType, String entityIdentificator) {
+        LOGGER.info("Create ACL for {}", entityType.name);
+        cli.createAcl(entityType, entityIdentificator, AclOperation.ALL, AclPermissionType.ALLOW, TOPIC_NAME);
+        var aclList = cli.listACLs(entityType, entityIdentificator);
+        Optional<AclBinding> aclFiltered = CLIUtils.searchAcl(aclList, entityIdentificator, AclOperation.ALL, AclPermissionType.ALLOW, AclResourceType.TOPIC, TOPIC_NAME);
+        assertTrue(aclFiltered.isPresent(), "ACL created must exist");
+    }
```

Comment:
this method could cleanup after itself and therefore don't need its priority. (It would not be big deal but we ar acutally using quite a lot of priorities in order to run these test as we expect, more explained in comment bellow)
```suggestion
    @Test(dataProvider = "aclEntityPairs")
    public void testCreateAllAllowAcl(ACLEntityType entityType, String entityIdentificator) {
        LOGGER.info("Create ACL for {}", entityType.name);
        cli.createAcl(entityType, entityIdentificator, AclOperation.ALL, AclPermissionType.ALLOW, TOPIC_NAME);
        Optional<AclBinding> aclFiltered = CLIUtils.searchAcl(cli, entityType, entityIdentificator, AclOperation.ALL, AclPermissionType.ALLOW, AclResourceType.TOPIC, TOPIC_NAME);
        assertTrue(aclFiltered.isPresent(), "ACL created must exist");

        //cleanup
        try {
            cli.deleteAcl(entityType, entityIdentificator, AclOperation.ALL, AclPermissionType.ALLOW);
        } catch (Exception ignore) { }
    }
```

Reply:
The problem about not setting test priorities is all creation tests are executed before all delete tests. If you prefer, to remove priorities and dependencies we can add pre-step to create ACLs for delete tests and post-step to delete ACLs for create tests.

Reply:
nope with proposed changes the only method that would stay would be first deleton method (i suggested to modify it that it as part of its setup also create given acl) => no longer depend on `createAcl` test